### ISSUE TITLE
Create and destroy monthly data sources and reports

### ISSUE THREAD

**Issue by adberglund** [project-koku/koku#1757]
## User Story
As a customer, I want older and unnecessary data to be cleaned up within operator metering, so that resources remain healthy and reports are generated and queryable without timeouts.

## Impacts
- Operator

## Assumptions
- OpenShift 4.2+ cluster
- Operator Metering has been installed and configured

## Implementation Details
Operator metering can run out of resources and report requests can time out due to large data sets. In order to combat this resource and timeout issue report objects can be more intelligently created and destroyed. Instead of creating reports without a start and end date we can create monthly report objects. Previous monthly report objects could be cleaned up several days into the next month. Currently we use a look back “window” of 10 days (we may need to shorten this to something like 3 days but it could also be configured. Worth considering if we can have the monthly report include the 3 days previous to the start of the month to continue to have an overlap to cover outages or missed reports. These reports would need to then be accessed via the API using a “monthly syntax” this could be as simple as using a YYYY-MM naming convention in the report object names.

## Acceptance Criteria
- [ ] Verify that report objects are created with monthly start and end periods
- [ ] Verify that report objects for a previous month are cleaned up several days into the following month

**Comment by nbon12** [comment]
Sprint Planning Meeting Notes:

Andrew: "As [it is] blocked by the previous issue. This requires a familiarity with the custom resources of the operator metering. Theoretically the knowledge around this is gained by the first or second issue when working on installing custom resources on operator metering."
